src/panel_gwalker/_gwalker.py

```diff
@@ -164,55 +430,177 @@ def _compute(self, payload):
-                'id': event_id,
-                'result': self._compute(msg['payload'])
-            })
-
-    async def export(
+        action = msg["action"]
+        event_id = msg.pop("id")
+        if action == "export" and event_id in self._exports:
+            self._exports[event_id] = msg["data"]
+        elif action == "compute":
+            self._send_msg(
+                {
+                    "action": "compute",
+                    "id": event_id,
+                    "result": self._compute(msg["payload"]),
+                }
+            )
+
+    async def _export(
         self,
-        mode: Literal['spec', 'svg'] = 'spec',
-        scope: Literal['current', 'all'] = 'current',
-        timeout: int = 5000
+        mode: Literal["spec", "svg", None] = None,
+        scope: Literal["current", "all", None] = None,
+        timeout: int | None = None,
     ):
         """
         Requests chart(s) on the frontend to be exported either
         as Vega specs or rendered to SVG.
 
         Arguments
         ---------
-        mode: 'code' | 'svg'
-           Whether to export the chart specification or SVG.
-        scope: 'current' | 'all'
-           Whether to export only the current chart or all charts.
-        timeout: int
-           How long to wait for the response before timing out.
+        mode: 'code' | 'svg' | None (default)
+           Whether to export the chart specification(s) or the SVG(s). If None the mode is set to the 'export_code' parameter value.
+        scope: 'current' | 'all' | None (default)
+           Whether to export only the current chart or all charts. If None the scope is set to the 'export_scope' parameter value.
+        timeout: int | None (default)
+           How long to wait for the response before timing out. If None the timeout is set to the 'export_timeout' parameter value.
 
         Returns
         -------
         Dictionary containing the exported chart(s).
         """
+        mode = mode or self.export_mode
+        scope = scope or self.export_scope
+        timeout = timeout or self.export_timeout
+
         event_id = uuid.uuid4().hex
-        self._send_msg({
-            'action': 'export',
-            'id': event_id,
-            'scope': f'{scope}',
-            'mode': mode
-        })
+        self._send_msg(
+            {"action": "export", "id": event_id, "scope": f"{scope}", "mode": mode}
+        )
         wait_count = 0
         self._exports[event_id] = None
         while self._exports[event_id] is None:
             await asyncio.sleep(0.1)
             wait_count += 1
             if (wait_count * 100) > timeout:
                 del self._exports[event_id]
-                raise TimeoutError(
-                    f'Exporting {scope} chart(s) timed out.'
-                )
+                raise TimeoutError(f"Exporting {scope} chart(s) timed out.")
         return self._exports.pop(event_id)
+
+    def create_export_settings(self, **params) -> Column:
```

Comment:
Note sure I like the `create_` naming. Also why are these separate methods? Seems like it'd be better as a combined `export_controls` method?

Reply:
For me its a convention that the `create_` suffix means that the object is `recreated` every time the function is run.

Without the prefix it is unclear and you have to look at docstring or  implementation to figure out.

Reply:
To me, in Python `create_` prefixes are kind of like `get_` prefixes, the fact that it's a method is already a signal to me that this is an operation that actually does something (like creating an object) and if it's simply returning an existing object then it should probably be a property or attribute.

Reply:
OK